server/mongo_data_access.go

```diff
@@ -226,7 +274,7 @@ func (dal *mongoDataAccessLayer) ConditionalDelete(query search.Query) (count in
 
 					if elementInSlice(id, successfulIds) {
 						// This resource was confirmed deleted
-						dal.invokeInterceptors("Delete", resourceType, elem.Resource)
+						dal.invokeInterceptorsAfter("Delete", resourceType, elem.Resource)
```

Comment:
Hmm, yeah I see what you mean. The error is not directly returned from the database operation (since some of the deletes clearly succeeded) but we've detected an error from the inconsistency. I'll add this change. Thanks!